Add image upload and blog edit information (#359)
unitaryfund/unitary.fund: README.md

year: 2023
---

# This is a heading style one

**This is a featured Paragraph it cannot contain bold text but can contain [links](https://www.unitary.fund).**

More text

::image[d394zg4utwqxynenamqf]

## This is a heading in heading style two

This is a standard paragraph with a [LINK](https://www.google.com). **This is bold paragraph text**

- this is a list item
- this is **bold** list item
- this is a long list item which has lots of words in it and will probably break onto a second or even a third line so we can see how that looks at differnt sizes adn make sure it looks as it is intended too
- this is a list item with a [LINK](https://www.google.com) or [two)](https://www.google.com) added in it. 

### This is a heading in heading style three introducing a video

> https://www.youtube.com/watch?v=dB_3R84ewig

Reviewer: I wonder if instead of demo-ing a subset of markdown here to act as a template, it might be more helpful to provide a link to more thorough documentation on how to use markdown.
- [GitHub docs](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax)
- [Write the Docs](https://www.writethedocs.org/guide/writing/markdown/)
- There's even a [markdown tutorial](https://www.markdowntutorial.com/)

Author: For me it is useful to show an example, as there are some elements specific here  as far as I can tell, e.g., 
```
**This is a featured Paragraph it cannot contain bold text but can contain [links](https://www.unitary.fund).**
```
or 
```
::image[d394zg4utwqxynenamqf]
```
Even if standard markdown, these prove quite useful to me even if I spent time previously editing markdown, due to how the website works.

Also in favor of having a template file, but could iterate in a later version.

#### This is a heading in heading style four

::gist[6cc14d37209de07abd707804f1b0219e]

With slashes

::gist[matt-lourens/6cc14d37209de07abd707804f1b0219e/]

> ::gist[/matt-lourens//6cc14d37209de07abd707804f1b0219e/]

Reviewer: I'd say we suggest a single one of these. Also it's unclear where `matt-lourens` is coming from . When uploading an image as described below, there is no user identifier that is embedded in the code. This either needs explanation, or removal.

Author: Good point. They seem to come from https://gist.github.com/matt-lourens/ and then the second is https://gist.github.com/matt-lourens/6cc14d37209de07abd707804f1b0219e.

> ::gist[6cc14d37209de07abd707804f1b0219e]
> 
> With slashes
> 
> ::gist[matt-lourens/6cc14d37209de07abd707804f1b0219e/]
> 
> ::gist[/matt-lourens//6cc14d37209de07abd707804f1b0219e/]

Author: ```suggestion

## Including code snippets with gists
A supported way to include code snippets is by embedding gists. Gists are code snippets that you can create on Github. Make sure that they are public. For example, navigating to [gist.github.com/matt-lourens/](https://gist.github.com/matt-lourens/) one can click on one of the gists and obtain the url  https://gist.github.com/matt-lourens/6cc14d37209de07abd707804f1b0219e. In order to embed the related code in the blog post, you can do one of the following:

::gist[6cc14d37209de07abd707804f1b0219e]

or with slashes

::gist[matt-lourens/6cc14d37209de07abd707804f1b0219e/]
```

```

Once your post is ready, open a pull request (PR)!

The Vercel automation will create a preview you can access by clicking the "Visit preview" link that shows up in the bot comment.
If the changes look good, request a review by tagging one of the UF team members (e.g., `@nathanshammah`).

## Adding images

To add an image in a page, there are two options:

> 1. (**No Code option**): Go to https://unitary-fund.vercel.app/author/image + copy the code (e..g, d394zg4utwqxynenamqf) + paste it in the markdown file as "::image[d394zg4utwqxynenamqf]". This option is the one required for adding the cover image to a new `event`.

Reviewer: The `+` here reads to me as string concatenation. I'd rephrase as something like
```
Visit https://unitary.... and upload your desired image.
Copy the code it returns after upload and use such and such syntax to insert.
```

> 2. Upload the image under `public/images/` + embed image links in markdown files e.g., ![(images/image.png), and embed it into the desired page, e.g., using the `![alt text](/images/your_image.png)` syntax.

Reviewer: I don't think the first syntax suggested in this change is correct.
```suggestion
2. Upload the image to `public/images/` and embed it into the desired page using the `![alt text](/images/your_image.png)` syntax.
```

### This is a heading in heading style three introducing a video

https://www.youtube.com/watch?v=dB_3R84ewig

#### This is a heading in heading style four

## Including code snippets with gists
A supported way to include code snippets is by embedding gists. Gists are code snippets that you can create on Github. Make sure that they are public. For example, navigating to [gist.github.com/matt-lourens/](https://gist.github.com/matt-lourens/) one can click on one of the gists and obtain the url  https://gist.github.com/matt-lourens/6cc14d37209de07abd707804f1b0219e. In order to embed the related code in the blog post, you can do one of the following:

::gist[6cc14d37209de07abd707804f1b0219e]

or with slashes

::gist[matt-lourens/6cc14d37209de07abd707804f1b0219e/]
```

Once your post is ready, open a pull request (PR)!

The Vercel automation will create a preview you can access by clicking the "Visit preview" link that shows up in the bot comment.
If the changes look good, request a review by tagging one of the UF team members (e.g., `@nathanshammah`).

## Adding images

To add an image in a page, there are two options:

> 1. (**No Code option**): Go to https://unitary-fund.vercel.app/author/image + copy the code (e..g, d394zg4utwqxynenamqf) + paste it in the markdown file as "::image[d394zg4utwqxynenamqf]". This option is the one required for adding the cover image to a new `event`.

Author: ```suggestion
1. (**No Code option**): Go to https://unitary-fund.vercel.app/author/image and upload your desired image. 
 Copy the code it returns after upload and use such and such syntax to insert. (e.g, d394zg4utwqxynenamqf). Note that this option is the required one for adding a cover image to a new `event`.
```

year: 2023
---

# This is a heading style one

> **This is a featured Paragraph it cannot contain bold text but can contain [links](https://www.unitary.fund).**

Reviewer: What flavor of markdown is this? Is a featured paragraph the same as a bolded paragraph?
```suggestion
**This is a featured paragraph it cannot contain bold text but can contain [links](https://www.unitary.fund).**
```

More text

::image[d394zg4utwqxynenamqf]

## This is a heading in heading style two

> This is a standard paragraph with a [LINK](https://www.google.com). **This is bold paragraph text**

Reviewer: ```suggestion
This is a standard paragraph with a [hyperlink](https://www.example.com). **This is bold paragraph text**.
```

- this is a list item
- this is **bold** list item

> - this is a long list item which has lots of words in it and will probably break onto a second or even a third line so we can see how that looks at differnt sizes adn make sure it looks as it is intended too

Reviewer: ```suggestion
- this is a long list item which has lots of words in it and will probably break onto a second or even a third line so we can make sure it looks as it is intended too. Can even have multiple sentences within a list item.
```

> - this is a list item with a [LINK](https://www.google.com) or [two)](https://www.google.com) added in it. 

Reviewer: ```suggestion
- this is a list item with a [link](https://www.example.com) or [two](https://www.example.com) added in it. 
```

> - this is a list item with a [LINK](https://www.google.com) or [two)](https://www.google.com) added in it. 

Author: ```suggestion
- this is a list item with a [link](https://unitaryhack.dev/) added in it. 
```

### This is a heading in heading style three introducing a video

> https://www.youtube.com/watch?v=dB_3R84ewig

Reviewer: Is this intended to be an embedded video or just a link? I'm not confident this will render as a clickable link (and I can't check as we do not have dev docs yet 😕).

Author: @natestemen yeah it's strange but it should render as an embedded frame. I just tested it [here](https://unitary-fund-prel-lj20ttudw-unitary-fund.vercel.app/posts/test) in [this PR](https://github.com/unitaryfund/unitary.fund/pull/367). I believe it is a special feature implemented by the designers, not a general property of md. 

I think the same applies for paragraph formatting, that is why I suggest keeping reference to it.

#### This is a heading in heading style four

## Including code snippets with gists

> A supported way to include code snippets is by embedding gists. Gists are code snippets that you can create on Github. Make sure that they are public. For example, navigating to [gist.github.com/matt-lourens/](https://gist.github.com/matt-lourens/) one can click on one of the gists and obtain the url  https://gist.github.com/matt-lourens/6cc14d37209de07abd707804f1b0219e. In order to embed the related code in the blog post, you can do one of the following:

Reviewer: ```suggestion
A supported way to include code snippets is by embedding gists.
Gists are code snippets that you can create on Github.
Learn how to create one [here](https://docs.github.com/en/get-started/writing-on-github/editing-and-sharing-content-with-gists/creating-gists), and don't forget to ensure it is public!
Once created, use the hash from the gist URL with the syntax below.
```

> ::gist[6cc14d37209de07abd707804f1b0219e]
> 
> or with slashes

Reviewer: ```suggestion
```
Let's save folks the decision and just provide one option.

::gist[matt-lourens/6cc14d37209de07abd707804f1b0219e/]
```

Once your post is ready, open a pull request (PR)!

The Vercel automation will create a preview you can access by clicking the "Visit preview" link that shows up in the bot comment.
If the changes look good, request a review by tagging one of the UF team members (e.g., `@nathanshammah`).

## Adding images

To add an image in a page, there are two options:

1. (**No Code option**): Go to https://unitary-fund.vercel.app/author/image and upload your desired image. 

>  Copy the code it returns after upload and use such and such syntax to insert. (e.g, d394zg4utwqxynenamqf). Note that this option is the required one for adding a cover image to a new `event`.

Author: ```suggestion
 Copy the code it returns after upload, e.g, d394zg4utwqxynenamqf, and use the syntax `::image[d394zg4utwqxynenamqf]` to insert it. Note that this option is the required one for adding a cover image to a new `event`.
```

>  Copy the code it returns after upload and use such and such syntax to insert. (e.g, d394zg4utwqxynenamqf). Note that this option is the required one for adding a cover image to a new `event`.

Reviewer: I think you may have copy and pasted my suggestion from https://github.com/unitaryfund/unitary.fund/pull/359#discussion_r1316369594 which was a template, but did not include all the details (which is why I did not use the suggestion tool). E.g. "such and such syntax" needs replaced.
